src/shared/utilities/timeoutUtils.ts

```diff
@@ -69,14 +84,16 @@ export class Timeout {
 
     /**
      * Kills the internal timer and resolves the timer's promise
+     * @param reject Rejects the token with a cancelled error message
      */
-    public killTimer(): void {
-        if (this.timerTimeout) {
-            clearTimeout(this.timerTimeout)
-        }
-        if (this.timerResolve) {
+    public killTimer(reject?: boolean): void {
```

Comment:
Since we are changing this class significantly as well as adding semantics to killTimer(), let's rename it (should have been named stop() originally, anyways). There are few existing usages so it will cause minimal churn.

```suggestion
    public stop(reject?: boolean): void {
```

Reply:
Rethinking: killTimer() appears to be resetting the Timeout. Maybe `complete()` is a good name.

Should refresh() throw an error if `this._fulfilled` is true? Or should it unset `this._fulfilled`?

Reply:
Stop and start functionality is a pretty good idea, though not needed yet.

Reply:
`clearTimeout` in `killTimer` completely stops the scheduled timer that would trigger the Timeout's expiration. `complete` could be a better name for it since it applies a sense of a finality. And yes refresh should throw an error.